src/dbnode/storage/series/buffer.go

```diff
@@ -930,12 +931,10 @@ func (b *BufferBucket) merge() (int, error) {
 		start   = b.start
 		readers = make([]xio.SegmentReader, 0, len(b.encoders)+len(b.bootstrapped))
 		streams = make([]xio.SegmentReader, 0, len(b.encoders))
-		iter    = b.opts.MultiReaderIteratorPool().Get()
```

Comment:
Am I missing something or was this `iter` completely unused?

Reply:
Yeah you're correct. I've fixed it locally, but of course it doesn't fix the issue haha.